x-pack/test/security_solution_api_integration/test_suites/investigation/saved_objects/tests/notes.ts

```diff
@@ -396,33 +371,33 @@ export default function ({ getService }: FtrProviderContext) {
         ]);
 
         const response = await supertest
-          .get('/api/note?sortField=eventId&sortOrder=desc')
+          .get(`${NOTE_URL}?sortField=eventId&sortOrder=desc`)
           .set('kbn-xsrf', 'true')
           .set('elastic-api-version', '2023-10-31');
-
-        const { totalCount, notes } = response.body;
+        const { notes, totalCount } = response.body as GetNotesResult;
 
         expect(totalCount).to.be(3);
         expect(notes[0].eventId).to.be('3');
         expect(notes[1].eventId).to.be('2');
         expect(notes[2].eventId).to.be('1');
       });
 
-      // TODO figure out why this test is failing on CI but not locally
-      it.skip('should retrieve all notes that have been created by a specific user', async () => {
-        await Promise.all([
-          createNote(supertest, { text: 'first note' }),
-          createNote(supertest, { text: 'second note' }),
-        ]);
-
-        const response = await supertest
-          .get('/api/note?userFilter=elastic')
-          .set('kbn-xsrf', 'true')
-          .set('elastic-api-version', '2023-10-31');
-
-        const { totalCount } = response.body;
-
-        expect(totalCount).to.be(2);
+      // this test cannot be run in serverless because in serverless the default user is 'test_user' instead of 'elastic'
+      describe('@skipInServerless', () => {
+        it('should retrieve all notes that have been created by a specific user', async () => {
+          await Promise.all([
+            createNote(supertest, { text: 'first note' }),
+            createNote(supertest, { text: 'second note' }),
+          ]);
+
+          const response = await supertest
+            .get(`${NOTE_URL}?userFilter=elastic`)
```

Comment:
I wonder if there's a way to use `/internal/security/me` to get the current user's username. That would enable this test in serverless as well.

Reply:
That is a great idea, let me try that tomorrow!

Reply:
As discussed in the thread, we are saving note with `full name` in `createdBy` field so this test will not work till we pass `full name` to `userFilter` of the active user  in the test.

Do you want to tackle that separately or in this PR itself?

Reply:
we discussed this in a Slack thread so I'm summarizing here. We have 2 options to tackle this:
- change the logic in the `persistNote` when saving/creating a note. We should not be using the `getUserDisplayName` function that can return `full_name`, `email` or `username`. Creating a note with a value that can change is not good, we should be using the same value always, and that value should be guaranteed to be there... This is out of scope for this PR
- modify the test to use `full_name` instead of `username`. I started working on this and could get it to work on `serverless` but now `ess` breaks... Adding a new utils function in `serverless` is easy as we can add a new utils function and retrieve whatever field we want like we do [here](https://github.com/elastic/kibana/blob/main/x-pack/test/security_solution_api_integration/config/services/security_solution_serverless_utils.ts#L65), but for `ess` it is trickier as we retrieve the values [this way](https://github.com/elastic/kibana/blob/main/x-pack/test/security_solution_api_integration/config/services/security_solution_ess_utils.ts#L22) and I'm not comfortable making changes to config files and such in this PR

I will keep this test skipped for now and will log a ticket for us to make one of the 2 changes above later!

Reply:
created [a ticket](https://github.com/elastic/kibana/issues/196896) and linked it in the test code [here](https://github.com/elastic/kibana/pull/196774/commits/351512599ac430461f6b9975564a7c1ce8ec0a55)